secure_message/common/utilities.py

```diff
@@ -143,6 +143,43 @@ def get_details_by_uuids(uuids):
 
 def add_to_and_from_details(messages):
     """Adds user details for sender and recipient"""
+    for message in messages:
+        message['@msg_from'] = _get_from_details(message)
+        message['@msg_to'] = _get_to_details(message)
+    return messages
+
+
+def _get_from_details(message):
+    """looks up the details for the from users"""
+    if message['from_internal']:
+        from_details, _ = internal_user_service.get_user_details(message['msg_from'])
+    else:
+        from_details, _ = party.get_user_details(message['msg_from'])
+    return from_details
+
+
+def _get_to_details(message):
+    """looks up the details of all the to users"""
+    user_details = []
+    if message['from_internal']:
+        to_user_details_finder = party.get_user_details
+    else:
+        to_user_details_finder = internal_user_service.get_user_details
+
+    for uuid in message['msg_to']:
+
+        detail, status_code = to_user_details_finder(uuid)
+
+        if status_code == 200:
+            user_details.append(detail)
+        else:
+            logger.info('No details found for user', uuid=uuid)
```

Comment:
I think this should be party_id rather than uuid as that doesn't really mean anything.

Reply:
God catch , this code should have been deleted , its been superseded . Deleting .

Reply:
Logging should be done as close to the code as possible. In this case, anything other than a 200 should be handled in to_user_details_finder. If you deal with HTTPErrors there, a lot of this function could be done as:

```python
[to_user_details_finder(uuid)[0] for uuid in message['msg_to']]
```